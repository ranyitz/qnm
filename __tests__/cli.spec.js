const { execSync } = require('child_process');
const { resolveFixture } = require('./utils');

const qnmBin = require.resolve('../bin/qnm');
const runCommand = (command, { cwd }) =>
  execSync(`${qnmBin} ${command}`, { cwd, encoding: 'utf-8' });

describe('CLI', () => {
  describe('qnm with no arguments', () => {
    it('should show the version on a single module when called with a string', () => {
      const cwd = resolveFixture('single-module');
      const output = runCommand('test', { cwd });

      expect(output).toMatchSnapshot();
    });

    it('should show get matchs when using the match command', () => {
      const cwd = resolveFixture('single-module');
      const output = runCommand('match te', { cwd });

      expect(output).toMatchSnapshot();
    });

    it('should add dependents information when using thw --why option', () => {
      const cwd = resolveFixture('single-module');
      const output = runCommand('--why test', { cwd });

      expect(output).toMatchSnapshot();
    });

    it('should add dependents information when using thw -w option', () => {
      const cwd = resolveFixture('single-module');
      const output = runCommand('-w test', { cwd });

      expect(output).toMatchSnapshot();
    });
  });

  describe('qnm list', () => {
    it('should show all modules in node_modules directory', () => {
      const cwd = resolveFixture('mix-modules');
      const output = runCommand('list', { cwd });

      expect(output).toMatchSnapshot();
    });

    it('should show modules mentioned in package.json', () => {
      const cwd = resolveFixture('indirect-dependencies');
      const output = runCommand('list --deps', { cwd });

      expect(output).toMatchSnapshot();
    });
  });
});
