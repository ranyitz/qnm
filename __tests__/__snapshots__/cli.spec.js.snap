// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI qnm list should show all modules in node_modules directory 1`] = `
"@scope/test
└── 1.0.0

another
└── 1.0.0

test
├── 1.0.0
└─┬ another
  └── 1.0.0

"
`;

exports[`CLI qnm list should show modules mentioned in package.json 1`] = `
"dependency1
└── 1.0.0

dependency2
└── 1.0.0

devDependency1
└── 1.0.0

devDependency2
└── 1.0.0

"
`;

exports[`CLI qnm with no arguments should add dependents information when using thw --why option 1`] = `
"test
└── 1.0.0 (devDependencies, npm install test)

"
`;

exports[`CLI qnm with no arguments should add dependents information when using thw -w option 1`] = `
"test
└── 1.0.0 (devDependencies, npm install test)

"
`;

exports[`CLI qnm with no arguments should show get matchs when using the match command 1`] = `
"test
└── 1.0.0

"
`;

exports[`CLI qnm with no arguments should show the version on a single module when called with a string 1`] = `
"test
└── 1.0.0

"
`;
